src/main/osd/osd_elements.c

```diff
@@ -1375,12 +1394,16 @@ void osdAnalyzeActiveElements(void)
     }
 #endif // GPS
 
-#ifdef USE_ESC_SENSOR
+#if defined(USE_ESC_SENSOR) || defined(USE_RPM_FILTER)
+    if ((featureIsEnabled(FEATURE_ESC_SENSOR)) || (motorConfig()->dev.useDshotTelemetry)) {
+        osdAddActiveElement(OSD_ESC_RPM);
+    } 
+    
     if (featureIsEnabled(FEATURE_ESC_SENSOR)) {
```

Comment:
This should not be in the same `#if` block now. `FEATURE_ESC_SENSOR` is only dependent on `USE_ESC_SENSOR` and not on `USE_RPM_FILTER`.

Reply:
This is fixed in the latest push